router/router.go

```diff
@@ -218,7 +218,7 @@ func New(cfg *config.Configuration, rateConvertor *currencies.RateConverter) (r
 	gdprPerms := gdpr.NewPermissions(context.Background(), cfg.GDPR, adapters.GDPRAwareSyncerIDs(syncers), theClient)
 
 	exchanges = newExchangeMap(cfg)
-	theExchange := exchange.NewExchange(theClient, pbc.NewClient(&cfg.CacheURL), cfg, r.MetricsEngine, bidderInfos, gdprPerms, rateConvertor)
+	theExchange := exchange.NewExchange(theClient, pbc.NewClient(&cfg.CacheURL, &cfg.ExtCacheURL), cfg, r.MetricsEngine, bidderInfos, gdprPerms, rateConvertor)
```

Comment:
Do we need to pass the `ExtCacheURL` to the cache client? Because it doesn't seem like the client itself uses the external cache URL for anything. Cache client is only interested in the internal URL to be able to connect to the cache. So in that case, any reason why we can't just directly pass `cfg.ExtCacheURL` as it is to the `NewExchange` function and then it can use it directly rather than trying to get it from the client like this: `e.cache.GetExtCacheData()`

Reply:
The reason we access the host and the path via `e.cache.GetExtCacheData()` is that `e.cache` is an interface so, to my knowledge we can't just do:
```
pathThatWillBeTargeted = e.cache.ExtCacheURL.Path
hostThatWillBeTargeted = e.cache.ExtCacheURL.Host
```
We must access that data via a function, hence, the need of declaring `GetExtCacheData() (string, string)`.
```
19 type Client interface {
20     // PutJson stores JSON values for the given openrtb.Bids in the cache. Null values will be
21     //
22     // The returned string slice will always have the same number of elements as the values argument. If a
23     // value could not be saved, the element will be an empty string. Implementations are responsible for
24     // logging any relevant errors to the app logs
25     PutJson(ctx context.Context, values []Cacheable) ([]string, []error)
26  +
27  +  // Serves the purpose of a getter that returns the host and the cache of the prebid-server URL
28  +  GetExtCacheData() (string, string)
29 }
prebid_cache_client/client.go
```
The `clientImpl` object implements `GetExtCacheData()` to extract those values like this
```
59 type clientImpl struct {
60     httpClient        *http.Client
61     putUrl            string
62  +  externalCacheHost string
63  +  externalCachePath string
64 }
65
66 + func (c *clientImpl) GetExtCacheData() (string, string) {
67 +     return c.externalCacheHost, c.externalCachePath
68 + }
prebid_cache_client/client.go
```
Test file `exchange/auction_test.go` defines `mockCache` to implement the `Cache` interface too but their `GetExtCacheData() (string, string)` gets implemented differently:
```
253 + func (c *mockCache) GetExtCacheData() (string, string) {
254 +     return "", ""
255 + }
256 func (c *mockCache) PutJson(ctx context.Context, values []prebid_cache_client.Cacheable) ([]string, []error) {
257     c.items = values
258     return []string{"", "", "", "", ""}, nil
259 }
exchange/auction_test.go
```

Reply:
Do you think there's a better way to access that data?

Reply:
Maybe it wasn't very clear from my comment but I didn't mean to say to access it via e.Cache. What I meant was that we, send `ExtCacheURL` directly to the `NewExchange` method and then store it in the exchange object and in `HoldAuction` method you can directly access it via the exchange object. Of course, you will have to modify the `Exchange` struct a little bit for that. 

But on further thought, your approach is also fine too. I just wish we called that interface `Cache` rather than `Client`.